src/plugins/data/common/index_patterns/index_patterns/index_pattern.ts

```diff
@@ -196,6 +195,34 @@ export class IndexPattern implements IIndexPattern {
     this.initFields();
   }
 
+  public initFromSpec(spec: IndexPatternSpec) {
```

Comment:
>At that point it would feel especially odd to have a dedicated initFromSpec, so I'd propose moving this into the constructor either here or at that time

100%. I wish to address this in a future PR.